GiniSwitchSDK/GiniSwitchSDK/Classes/GiniSwitchSdk.swift

```diff
@@ -56,26 +48,72 @@ public class GiniSwitchSdk {
         self.clientId = clientId
         self.clientSecret = clientSecret
         self.clientDomain = domain
-        GiniSwitchSdkStorage.activeSwitchSdk = self     // TODO: Don't use GiniSwitchSdkStorage - find a better solution
+        extractionsManager = ExtractionsManager(clientId: clientId, clientSecret: clientSecret, clientDomain: clientDomain)
+        extractionsManager.delegate = self
     }
     
     /*
      * Convenience method for creating the SDK's UI
      */
     public func instantiateSwitchViewController() -> UIViewController {
-        return userInterface.initialViewController
+        coordinator = MultiPageCoordinator(extractionsManager: extractionsManager, onboarding: configuration.onboarding)
+        coordinator?.delegate = self
+        return coordinator!.initialViewController
     }
     
     public func sendFeedback(_ feedback:ExtractionCollection) {
-        feedbackHandler(feedback)
+        extractionsManager.sendFeedback(feedback)
+    }
+
+}
+
+extension GiniSwitchSdk: ExtractionsManagerDelegate {
+    
+    func extractionsManager(_ manager:ExtractionsManager, didEncounterError error:NSError) {
+        // check if it's an error that needs to be shown to the user
+        if error.isHumanReadable {
```

Comment:
Why? If the error is human readable, it should be shown. If it's not, users will not be able to make sense of it, so we shouldn't show it at all.